### PR TITLE
enh(check + engine) : add class for check_result, use more shared_ptr.

### PR DESCRIPTION
**Pull request by SylvestreG**
add class for check_result, use more shared_ptr.

### CODE REVIEW

File: modules/external_commands/src/commands.cc [hunk 1]
```diff
@@ -923,12 +899,9 @@ int cmd_remove_acknowledgement(int cmd, char* args) {
     return ERROR;
 
   /* verify that the host is valid */
-  temp_host = nullptr;
   umap<uint64_t, std::shared_ptr<com::centreon::engine::host>>::const_iterator
     it(configuration::applier::state::instance().hosts().find(get_host_id(host_name)));
-  if (it != configuration::applier::state::instance().hosts().end())
-    temp_host = it->second.get();
-  if (temp_host  == nullptr)
+  if (it == configuration::applier::state::instance().hosts().end() || it->second.get() == nullptr)
```
Comment (SylvestreG) [inline modules/external_commands/src/commands.cc]: need to remove the get here.

File: modules/external_commands/src/commands.cc [hunk 2]
```diff
@@ -832,12 +812,9 @@ int cmd_acknowledge_problem(int cmd, char* args) {
...
```
Comment (SylvestreG) [inline modules/external_commands/src/commands.cc]: need to remove the get here.

File: modules/external_commands/src/commands.cc [hunk 3]
```diff
@@ -1019,10 +989,9 @@ int cmd_schedule_downtime(int cmd, time_t entry_time, char* args) {
     temp_host = nullptr;
     umap<uint64_t, std::shared_ptr<com::centreon::engine::host>>::const_iterator
       it(configuration::applier::state::instance().hosts().find(get_host_id(host_name)));
-    if (it != configuration::applier::state::instance().hosts().end())
-      temp_host = it->second.get();
-    if (temp_host  == nullptr)
+    if (it == configuration::applier::state::instance().hosts().end() || it->second.get() == nullptr)
```
Comment (SylvestreG) [inline modules/external_commands/src/commands.cc]: need to remove the get here.

File: modules/external_commands/src/commands.cc [hunk 4]
```diff
@@ -1212,10 +1181,9 @@ int cmd_schedule_downtime(int cmd, time_t entry_time, char* args) {
       umap<uint64_t, std::shared_ptr<com::centreon::engine::host>>::const_iterator
         found(configuration::applier::state::instance().hosts().find(
           get_host_id(it->first.first)));
-      if (found != configuration::applier::state::instance().hosts().end())
-        temp_host = found->second.get();
-      if (temp_host  == nullptr)
+      if (found == configuration::applier::state::instance().hosts().end() || found->second.get() == nullptr)
```
Comment (SylvestreG) [inline modules/external_commands/src/commands.cc]: same here.

File: src/checks/checker.cc [hunk 5]
```diff
@@ -389,24 +388,7 @@ void checker::run(
...
-  check_result_info.return_code = notifier::ok;
-  check_result_info.output = nullptr;
-  check_result_info.output_file_fd = -1;
-  check_result_info.output_file_fp = nullptr;
-  check_result_info.output_file = nullptr;
-  check_result_info.host_name = string::dup(hst->get_name());
-  check_result_info.service_description = nullptr;
-  check_result_info.latency = latency;
-  check_result_info.next = nullptr;
+
```
Comment (SylvestreG) [inline src/checks/checker.cc]: check_result_info.check_type = check_active;	
  check_result_info.check_options = check_options;	
  check_result_info.scheduled_check = scheduled_check;	
  check_result_info.reschedule_check = reschedule_check;	
  check_result_info.start_time = start_time;	
  check_result_info.finish_time = start_time;	
  check_result_info.early_timeout = false;	
  check_result_info.exited_ok = true;	
  check_result_info.return_code = notifier::ok;	
  check_result_info.output = nullptr;	
  check_result_info.output_file_fd = -1;	
  check_result_info.output_file_fp = nullptr;	
  check_result_info.output_file = nullptr;	
  check_result_info.host_name = string::dup(hst->get_name());	
  check_result_info.service_description = nullptr;	
  check_result_info.latency = latency;	
  check_result_info.next = nullptr;

File: inc/com/centreon/engine/flapping.hh [hunk 6]
```diff
@@ -39,13 +39,13 @@ void enable_flap_detection_routines();
 // disables flap detection on a program-wide basis
 void disable_flap_detection_routines();
 // enables flap detection for a particular host
-void enable_host_flap_detection(com::centreon::engine::host* hst);
+void enable_host_flap_detection(std::shared_ptr<com::centreon::engine::host> hst);
```
Comment (bouda1) [inline inc/com/centreon/engine/flapping.hh]: This function could become a method and moved into host class.
As we should also have service with a similar one, I can expect that this code is moved into notifer class.

File: inc/com/centreon/engine/flapping.hh [hunk 7]
```diff
@@ -39,13 +39,13 @@ void enable_flap_detection_routines();
...
 // disables flap detection for a particular host
-void disable_host_flap_detection(com::centreon::engine::host* hst);
+void disable_host_flap_detection(std::shared_ptr<com::centreon::engine::host> hst);
```
Comment (bouda1) [inline inc/com/centreon/engine/flapping.hh]: Idem for this one.

File: inc/com/centreon/engine/flapping.hh [hunk 8]
```diff
@@ -39,13 +39,13 @@ void enable_flap_detection_routines();
...
 // enables flap detection for a particular service
-void enable_service_flap_detection(com::centreon::engine::service* svc);
+void enable_service_flap_detection(std::shared_ptr<com::centreon::engine::service> svc);
```
Comment (bouda1) [inline inc/com/centreon/engine/flapping.hh]: And here are the service ones.

File: inc/com/centreon/engine/flapping.hh [hunk 9]
```diff
@@ -39,13 +39,13 @@ void enable_flap_detection_routines();
...
 // disables flap detection for a particular service
-void disable_service_flap_detection(com::centreon::engine::service* svc);
+void disable_service_flap_detection(std::shared_ptr<com::centreon::engine::service> svc);
```
Comment (bouda1) [inline inc/com/centreon/engine/flapping.hh]: Idem

File: src/host.cc [hunk 10]
```diff
@@ -1446,32 +1446,32 @@ int host::handle_async_check_result_3x(check_result* queued_check_result) {
...
       << "\tCheck Type:         "
-      << (queued_check_result->check_type == check_active ? "Active"
+      << (queued_check_result->get_check_type() == check_active ? "Active"
                                                           : "Passive")
       << "\n"
-      << "\tCheck Options:      " << queued_check_result->check_options << "\n"
+      << "\tCheck Options:      " << queued_check_result->get_check_options() << "\n"
       << "\tScheduled Check?:   "
-      << (queued_check_result->scheduled_check ? "Yes" : "No") << "\n"
+      << (queued_check_result->get_reschedule_check() ? "Yes" : "No") << "\n"
```
Comment (SylvestreG) [inline src/host.cc]: remove scheduled_check here.

File: src/service.cc [hunk 11]
```diff
@@ -1075,26 +1076,26 @@ int service::handle_async_check_result(check_result* queued_check_result) {
   logger(dbg_checks, more)
       << "HOST: " << _hostname
       << ", SERVICE: " << _description << ", CHECK TYPE: "
-      << (queued_check_result->check_type == check_active ? "Active"
+      << (queued_check_result->get_check_type() == check_active ? "Active"
                                                           : "Passive")
-      << ", OPTIONS: " << queued_check_result->check_options << ", SCHEDULED: "
-      << (queued_check_result->scheduled_check ? "Yes" : "No")
+      << ", OPTIONS: " << queued_check_result->get_check_options() << ", SCHEDULED: "
+      << (queued_check_result->get_reschedule_check() ? "Yes" : "No")
```
Comment (SylvestreG) [inline src/service.cc]: remove scheduled_check here

File: src/checks/checker.cc [hunk 12]
```diff
@@ -811,7 +791,7 @@ void checker::run_sync(
   hst->set_latency(0.0);
 
   // Adjust check attempts.
-  adjust_host_check_attempt_3x(hst, true);
+  hst->adjust_check_attempt_3x(true);
```
Comment (bouda1) [inline src/checks/checker.cc]: This method looks like the service one ::adjust_check_attempt(), maybe we could harmonize their names.

**Pull request merged**